api/v1alpha3/machine_webhook.go

```diff
@@ -54,10 +67,33 @@ func (m *Machine) validate() error {
 	if m.Spec.Bootstrap.ConfigRef == nil && m.Spec.Bootstrap.Data == nil {
 		allErrs = append(
 			allErrs,
-			field.Required(field.NewPath("spec", "bootstrap", "data"), "expected spec.bootstrap.data or spec.bootstrap.configRef to be populated"),
+			field.Required(
+				field.NewPath("spec", "bootstrap", "data"),
+				"expected spec.bootstrap.data or spec.bootstrap.configRef to be populated",
+			),
 		)
+	}
 
+	if m.Spec.Bootstrap.ConfigRef != nil && m.Spec.Bootstrap.ConfigRef.Namespace != m.Namespace {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "bootstrap", "configRef", "namespace"),
+				"must match metadata.namespace",
+			),
+		)
 	}
+
+	if m.Spec.InfrastructureRef.Namespace != m.Namespace {
+		allErrs = append(
+			allErrs,
+			field.Forbidden(
+				field.NewPath("spec", "infrastructureRef", "namespace"),
```

Comment:
One thing I wanted to point out here is that I'm specifying `infrastructureRef` and `configRef` rather than `infrastructureref` and `configref`. I'm not sure what the convention is but I thought having it caml case made it a little more readable.

Reply:
It's supposed to be camel case, leading letter lowercased, to match the OpenAPI field definitions.

Reply:
Nice! I like when something is intuitive like that! Thanks for confirming.